src/core/util_sndfile.c

```diff
@@ -31576,7 +31576,8 @@ wav_read_header	 (SF_PRIVATE *psf, int *blockalign, int *framesperblock)
 						}
 					else if (psf->filelength < RIFFsize + 2 * SIGNED_SIZEOF (dword))
 					{	psf_log_printf (psf, "RIFF : %u (should be %D)\n", RIFFsize, psf->filelength - 2 * SIGNED_SIZEOF (dword)) ;
-						RIFFsize = dword ;
+						dword = psf->filelength - 2 * sizeof (dword) ;
```

Comment:
What is this doing?

Reply:
Dusty braincells.  Seems to me it snuck in with the Midi fixes and was part of my separate task of diagnosing that SndFileOut doesn't properly cleanup/close files during a chuck exit (iirc).  Also, I've been meaning to revisit the MIDI issue because I notice that I can get spurious crashes when ctrl-c exiting ChucK when a MIDI device is open.  So the change is likely to be incomplete. (in addition to this bit of sloppiness).

Reply:
also of note: 9a487e6a6bb1e422a9ead9e69e31ff7b0cad6fad
indicates that in this condition dword  is unitialized/meaningless, so that's probably the real point here.

Reply:
gotcha for the sndfile stuff. I"ll see if it's feasible to unit test that.

Regarding the ctrl-c issues, I think we may have encountered similar problems (see #215 ).  I think it's an issue with events in general and not just midiIn, but if you have any insights into that please leave a comment there.

Either way this pr does improve stability for midiIn so I think it's worth including while that gets investigated.

Reply:
sounds good!